### ISSUE TITLE
Comment misidentified as bad HLINT pragma

### ISSUE THREAD

**Issue by sjakobi** [ndmitchell/hlint#658]
Given

```
main = do {- Lint -} x
```

`hlint` reports

```
≻ hlint bug.hs 
bug.hs:1:11: Error while reading hint file, bad HLINT pragma, expected:
    {- HLINT <severity> <identifier> "Hint name" -}
{- Lint -}
```

**Comment by ndmitchell** [comment]
Hmm... I wonder if the answer here is to say that for HLint users you shouldn't write comments named Lint, and should expand on them a bit, since not detecting it risks missing invalid pragmas and ignoring them silently. Does that Lint refer to HLint? I really want to use `{-# HLINT #-}` but GHC HQ rejected my patch, so that's not an option...

**Comment by sjakobi** [comment]
I reduced the reproduction from  [this code](https://gitlab.haskell.org/ghc/ghc/blob/ba4e3934abc82e0ba2bec51842315819910d1018/compiler/main/HscMain.hs#L1842).

Interestingly `hlint` will only throw the error if it finds the comment in certain positions. So for example

```
main = x {- Lint -}
```

or 

```
{- Lint -}
main = x
```

don't trigger the error.

**Comment by ndmitchell** [comment]
Very interesting! I think those two might be the real bugs!

**Comment by ndmitchell** [comment]
The issue was that if you had no hints then HLint didn't demand the classifiers list, and the classifiers list ended up with errors inside it. I now force the classifiers which means they are all errors.

Since GHC HQ decided to reject my proposal to allow `{-# HLINT #-}`, pretty much forcing me to lex comments, I suggest a small patch to GHC to avoid using that particular comment. Naming it to `Run lint if necessary` captures the same ideas. If that's infeasible, then I could try turning the invalid comments into hints themselves, which would mean they could be captured and ignored...

**Comment by sjakobi** [comment]
Interesting!

Changing that comment in GHC shouldn't be a problem.

But is the general message now, that HLint users mustn't write `{- -}` comments that start with the word `Lint`? I understand that you want to catch typoed `HLINT` pragmas but I suspect that the large majority of comments that start with `Lint` will be false positives. Visually, it's hard to confuse `Lint` or `lint` with `Hlint` or `hlint`.

Wouldn't it be better to make an exception for comments that start with `Lint` or `lint`?

**Comment by ndmitchell** [comment]
Good point - I supported both HLINT and LINT, but never documented LINT, and it does seem to be annoying in practice, so I've made it require only HLINT.

**Comment by sjakobi** [comment]
Awesome. Thank you! :)
